### DISCUSSION TITLE
New Feature Idea: Allow blueprints/automation to create helper entities...

### DISCUSSION THREAD

**Discussion by jeeftor** [home-assistant/architecture#935] [General] [answered]
(I wasn't sure where this goes).

I wanted to suggest the option to define a set of helper entities within a blueprint that could be used by that blueprint. Thus by allowing blueprints to maintain state.

In a blueprint you can define inputs - perhaps you could define helpers that could be used by the blueprints.

So if an automation is created:

```
automation.my_automation
```

perhaps you could have:

```
number.automation_my_automation
```

or something like that.  I think it could make it easier for blueprints that require helpers to be installed by users.

**Comment by frenck** [comment, marked as answer]
Right, so this has been an idea for a long time already. However, architectural discussions are not about sharing ideas; they are about making architectural implementation proposals and getting them approved so you can continue with the implementation.

Could you please add a technical proposal, that sets the context, the proposal, possible consequences and details on the implementation details?

../Frenck
